snuba/utils/streams/kafka.py

```diff
@@ -35,6 +36,27 @@ class TransportError(ConsumerError):
 
 
 class KafkaConsumer(Consumer[TopicPartition, int, bytes]):
+    """
+    The behavior of the of this consumer differs slightly from the Confluent
+    consumer during rebalancing operations. Whenever a partition is assigned
+    to this consumer, offsets are *always* automatically reset to the
+    committed offset for that partition (or if no offsets have been committed
+    for that partition, the offset is reset in accordance with the
+    ``auto.offset.reset`` configuration value.) This causes partitions that
+    are maintained across a rebalance to have the same offset management
+    behavior as a partition that is moved from one consumer to another. To
+    prevent uncommitted messages from being consumed multiple times,
+    ``commit`` should be called in the partition revocation callback.
```

Comment:
Isn't this a problem for the batching consumer that is supposed to commit only when a batch is flushed ?